vault/src/system.rs

```diff
@@ -553,10 +557,12 @@ impl VaultService {
     async fn maybe_register_vault(&self) -> Result<(), Error> {
         let account_id = self.btc_parachain.get_account_id();
 
-        let collateral_currency = if let Some(currency_id) = self.config.collateral_currency_id {
-            currency_id
+        // TODO: convert auto_register to Vec<(CurrencyId, u128)>
+        let (collateral_currency, collateral_amount) = if let Some((currency, amount)) = self.config.auto_register {
+            (currency, amount)
         } else {
-            self.btc_parachain.relay_chain_currency_id
+            // nothing to do
+            return Ok(());
```

Comment:
wouldn't this bail if you use faucet?

Reply:
Ah no because my idea was that the operator should use the KEY=value parsing in the first place but then we instead use the faucet if `faucet_url` is defined. I'm not sure if I'm 100% happy with this API yet though, just trying to figure out a way to limit user error when starting a vault.

Reply:
I don't follow? If you use the the faucet, `auto_register` would be None, so you'd return here with `Ok(())` rather than continuing to execute `fund_and_register` below

Reply:
I'm trying to find a nicer UX where those cli flags are complementary. So if you want to auto-register the vault we define `--auto-register=KSM=1000` but if you want to additionally use the faucet we also set `--faucet-url=xxx` and that auto-registers *with* the faucet.

Reply:
I see. To be honest I think it's more confusing to require both args. You'd do `--auto-register=KSM=1000 --faucet-url=xxx` but the 1000 is completely ignore in this case

Reply:
Maybe we can allow something like `--auto-register=KSM=from-faucet`? 

ALso, can you update the PR to take a vec rather than an option?